app/routes/github.js

```diff
@@ -0,0 +1,42 @@
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+const {
+  get,
+  inject: { service },
+  Route
+} = Ember;
+
+const CODE_INVALID = 'The GitHub authorization code is invalid.';
+
+export default Route.extend(AuthenticatedRouteMixin, {
+  queryParams: { code: { refreshModel: true } },
```

Comment:
You can see a lengthy bit of code that shows most of the inner workings here https://github.com/emberjs/ember.js/blob/v2.12.0/packages/ember-routing/lib/system/route.js#L171